### PR TITLE
Export ConnectOptions type from package

### PR DESCRIPTION
**Pull request by JKobrynski**
### Details

Export `ConnectOptions` type from the library. It turned out during ongoing Expensify TypeScript migration that this type needs to be exported from the library.

### Related Issues

[#24783](https://github.com/Expensify/App/issues/24783)

### CODE REVIEW

File: lib/Onyx.d.ts [hunk 1]
```diff
@@ -20,41 +19,6 @@ type NullableKeyValueMapping = {
     [TKey in OnyxKey]: OnyxEntry<KeyValueMapping[TKey]>;
 };
 
-/**
```
Comment (fabioh8010) [inline lib/Onyx.d.ts:23]: I think you don't need to move these types to other file, just export from here (like it was done for `OnyxUpdate`).

Comment (JKobrynski) [reply lib/Onyx.d.ts]: Thanks, done!

**Comment by JKobrynski** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by JKobrynski** [comment]
recheck

**Review by blazejkustra — APPROVED** [review]

**Review by arosiclair — APPROVED** [review]

**Pull request merged**